app/controllers/receive_package.js

```diff
@@ -121,85 +146,124 @@ export default Ember.Controller.extend({
 
   invalidInventoryNo: Ember.computed({
     get: function() {
-      var isValid = this.verifyInventoryNumber(this.get("package.inventoryNumber"));
+      var isValid = this.verifyInventoryNumber(
+        this.get("package.inventoryNumber")
+      );
       return isValid;
     },
     set: function(key, value) {
       return value;
     }
   }),
 
+  receivePackageParams() {
+    const pkgData = this.get("packageForm");
+    const inventoryNumber = this.get("inventoryNumber");
+    let pkg = this.get("package");
+    const locationId = this.get("locationId.id") || this.get("locationId");
+    if (locationId) {
+      var location = this.get("store").peekRecord("location", locationId);
+      pkg.set("location", location);
+    }
+    pkg.set("state", "received");
+    pkg.set("state_event", "mark_received");
+    pkg.set("quantity", pkgData.quantity);
+    pkg.set("length", pkgData.length);
+    pkg.set("width", pkgData.width);
+    pkg.set("height", pkgData.height);
+    pkg.set("notes", pkgData.notes);
+    pkg.set("inventoryNumber", inventoryNumber);
+    pkg.set("grade", this.get("selectedGrade.id"));
+    pkg.set("donorCondition", this.get("selectedCondition"));
+    return pkg;
+  },
+
+  isPackageValid() {
+    const pkgData = this.get("packageForm");
+    const inventoryNumber = this.get("inventoryNumber");
+    this.set("invalidQuantity", pkgData.quantity.toString().length === 0);
```

Comment:
We cannot be modifying properties inside getters like "isPackageValid", the name implies that it's a read-only property.

to do that, `invalidQuantity` should never be set, instead it should listen to `package.quantity` : 
```javascript
invalidQuantity: Ember.computed('package.quantity', function () {
   return this.get("package.quantity").length === 0;
})
```
similar to `invalidInventoryNo` above
Making all those properties reactive will also remove the need to do `this.notifyPropertyChange("watchErrors")`

and eventually this function will look like 
```
return !this.get('hasErrors') && !this.get("invalidInventoryNo");
```